shared/android/c++/Binding.cpp

```diff
@@ -0,0 +1,140 @@
+#include "Binding.h"
+
+#include <dlfcn.h>
+// #include <forandroid.h>
+#include <android/log.h>
+#include <android/trace.h>
+#include <jsi/JSIDynamic.h>
+#include <chrono>  // std::chrono::seconds
+#include <thread>
+#if __i386__
+#include <keybaselib-386.h>
+#elif __arm__
+#include <keybaselib-arm.h>
+#else
+#include <keybaselib-arm64.h>
+#endif
+
+// #ifndef __ANDROID_API__
+// #define __ANDROID_API__ 29
+// #endif
+
+#define APPNAME "MyApp"
+
+struct EventHandlerWrapper {
+  EventHandlerWrapper(jsi::Function eventHandler)
+      : callback(std::move(eventHandler)) {}
+
+  jsi::Function callback;
+};
+
+#if ANDROID
+extern "C" {
+JNIEXPORT void JNICALL Java_io_keybase_ossifrage_MainActivity_install(
```

Comment:
This is called from Java to install our JSI bindings into the react native runtime.